(FACT-2029) Solaris release facts

alexandru-d-popa opened the pull request.

alexandru-d-popa committed:
Merge remote-tracking branch 'origin/master' into FACT-2029

alexandru-d-popa committed:
(FACT-2029) Full os fact for Solaris

Added os release resolver implementation and release fact for solaris.

alexandru-d-popa committed:
(FACT-2029) Remove inspection profile.xml

Removed inspection profile.xml from the proj that was generated by rubymine

alexandru-d-popa committed:
(FACT-2029) Add tests for solaris os release

Added tests for solari os release resolver, and for release fact

sebastian-miclea commented on spec/facter/resolvers/solaris/solaris_release_resolver.rb:
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+describe 'SolarisOsReleaseResolver' do
```
I don't think these specs were run.

sebastian-miclea commented on lib/resolvers/solaris/solaris_os_release_resolver.rb:
```diff
@@ -0,0 +1,55 @@
+
+module Facter
+  module Resolvers
+    class SolarisOsRelease < BaseResolver
```
Could you rename this class to **SolarisRelease**? The direction we're heading towards is to have the resolver name describe as much as possible on what it checks, like _lsb_release_resolver_.

oanatmaria commented on lib/resolvers/solaris/solaris_os_release_resolver.rb:
```diff
@@ -0,0 +1,53 @@
+
+        def build_release_facts(fact_name)
+          result = read_os_release_file
+          return nil if result.nil?
```
just return

oanatmaria commented on lib/resolvers/solaris/solaris_os_release_resolver.rb:
```diff
@@ -0,0 +1,53 @@
+          output, error = Open3.capture2('cat /etc/release')
+          if error
+            @log.error('Could not build release fact because of missing file /etc/release')
+            return nil
```
just return

alexandru-d-popa committed:
Merge remote-tracking branch 'origin/master' into FACT-2029

oanatmaria commented on spec/facter/resolvers/solaris/solaris_release_resolver_spec.rb:
```diff
@@ -0,0 +1,63 @@
+    allow(Open3).to receive(:capture2)
+      .with('cat /etc/release')
+      .and_return([output, status])
+    # allow_any_instance_of(Process::Status).to receive(:to_s).and_return(st)
```
delete commented line

alexandru-d-popa committed:
(FACT-2029) Default value in case minor is missing

added default value of zero in case that minor version is not in the release file but a regex pattern is found.
